packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/SurroundingPairScopeHandler/SurroundingPairScopeHandler.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Direction,
+  Position,
+  SurroundingPairScopeType,
+  TextEditor,
+  showError,
+  type ScopeType,
+} from "@cursorless/common";
+import type { LanguageDefinitions } from "../../../../languages/LanguageDefinitions";
+import { BaseScopeHandler } from "../BaseScopeHandler";
+import { compareTargetScopes } from "../compareTargetScopes";
+import { TargetScope } from "../scope.types";
+import { ScopeIteratorRequirements } from "../scopeHandler.types";
+import { getDelimiterOccurrences } from "./getDelimiterOccurrences";
+import { getDelimiterRegex } from "./getDelimiterRegex";
+import { getIndividualDelimiters } from "./getIndividualDelimiters";
+import { getSurroundingPairOccurrences } from "./getSurroundingPairOccurrences";
+import { ide } from "../../../../singletons/ide.singleton";
+import { createTargetScope } from "./createTargetScope";
+import { stronglyContains } from "./stronglyContains";
+
+export class SurroundingPairScopeHandler extends BaseScopeHandler {
+  public readonly iterationScopeType: ScopeType = { type: "document" };
+  protected isHierarchical = true;
+
+  constructor(
+    private languageDefinitions: LanguageDefinitions,
+    public readonly scopeType: SurroundingPairScopeType,
+    private languageId: string,
+  ) {
+    super();
+  }
+
+  *generateScopeCandidates(
+    editor: TextEditor,
+    position: Position,
+    direction: Direction,
+    hints: ScopeIteratorRequirements,
+  ): Iterable<TargetScope> {
+    const individualDelimiters = getIndividualDelimiters(
+      this.scopeType.delimiter,
+      this.languageId,
```

Comment:
What about this scenario?
```ts
`foo ${value} bar`
```

Reply:
Huh interesting point. I also thought about this some more and a) doing something better here is almost certainly not worth the work and b) it might actually be nice if behaviour were consistent on commented code even though it's technically textual

Reply:
That is probably useful as well

Reply:
yeah let's leave it for now. I think it would be a lot more complex to get this one perfect so let's not waste the effort until we decide it's actually necessary